From 9b64398e6e850f2215b35fb40362e8e61b6c56e9 Mon Sep 17 00:00:00 2001
From: Antonios Vamporakis <ant@area128.com>
Date: Sun, 5 Jan 2014 03:00:08 +0100
Subject: MIPS: lantiq: lantiq_nand temp fix for SPL

SPL's NAND implementation does not call nand_scan which adds the read_buf method.
This fix will work for 8bit chips by adding the method in the driver but a proper
solution is needed.

Signed-off-by: Antonios Vamporakis <ant@area128.com>

diff --git a/drivers/mtd/nand/lantiq_nand.c b/drivers/mtd/nand/lantiq_nand.c
index 85f8f68..d444dea 100644
--- a/drivers/mtd/nand/lantiq_nand.c
+++ b/drivers/mtd/nand/lantiq_nand.c
@@ -98,6 +98,15 @@ static void ltq_nand_cmd_ctrl(struct mtd_info *mtd, int cmd, unsigned int ctrl)
 	}
 }
 
+static void ltq_nand_read_buf(struct mtd_info *mtd, uint8_t *buf, int len)
+{
+	int i;
+	struct nand_chip *chip = mtd->priv;
+
+	for (i = 0; i < len; i++)
+		buf[i] = readb(chip->IO_ADDR_R);
+}
+
 int ltq_nand_init(struct nand_chip *nand)
 {
 	/* Enable NAND, set NAND CS to EBU CS1, enable EBU CS mux */
@@ -108,6 +117,7 @@ int ltq_nand_init(struct nand_chip *nand)
 	nand->dev_ready = ltq_nand_dev_ready;
 	nand->select_chip = ltq_nand_select_chip;
 	nand->cmd_ctrl = ltq_nand_cmd_ctrl;
+	nand->read_buf = ltq_nand_read_buf;
 
 	nand->chip_delay = 30;
 	nand->options = 0;
-- 
2.7.4

